feature-pack/src/main/resources/modules/system/layers/base/org/wildfly/extension/grpc/main/module.xml

```diff
@@ -35,14 +35,14 @@
         <module name="org.jboss.staxmapper" />
         <module name="org.wildfly.security.manager" />
         <module name="io.netty.netty-common" />
-        <module name="io.netty.netty-codec" optional="true" />
-        <module name="io.netty.netty-handler" optional="true" />
+        <module name="io.netty.netty-codec" />
+        <module name="io.netty.netty-handler" />
         <!--
             Additional module dependencies.
             If they contain any classes that should be exposed to the user's code,
             (e.g. in a deployment) make sure you add them in the subsystem's
             DependencyProcessor too
         -->
-        <module name="org.wildfly.grpc-dependency" />
+        <module name="org.wildfly.grpc-dependency" export="true"/>
```

Comment:
Well, I was getting a "class not found" for something in com.google.protobuf, which is in org.wildfly.grpc-dependency, so this fixed it. I think anyone using grpc will need com.google.protobuf, so it seemed reasonable. 

What about moving com.google.protobuf to org.wildfly.extension.grpc and exporting it from there?

To be honest, I'm not sure why the modules are split into org.wildfly.extension.grpc and org.wildfly.grpc-dependency. I tried merging them into org.wildfly.extension.grpc but ran into trouble so I let it go.

I guess the comment "Additional module ..." is telling me to figure out exactly which jars are likely to be needed in user code and make them available by way of GrpcDependencyProcessor. Is that instead of exporting from module.xml or in addition to?

Reply:
What it sounds like is `com.google.protobuf` needs to be a module dependency here. I don't think exporting is right. I'd need to analyze the modules to see what should be added where. However, very generally speaking if a module needs to be exported it likely means something is configured wrong.